### PR TITLE
Transcode hevc h2s

### PR DESCRIPTION
**Pull request by xhaihao**
This is used to make sure transcode + tonemaping works as expected.

### CODE REVIEW

**Comment by xhaihao** [comment]
The output from tonemap_vaapi still has HDR side data. Fixed in https://github.com/intel-media-ci/ffmpeg/pull/664

**Comment by xhaihao** [comment]
@uartie  @Bin-CI @FocusLuo Any comment for this PR ?

**Comment by FocusLuo** [comment]
> @uartie @Bin-CI @FocusLuo Any comment for this PR ?

@xhaihao I gave a comment for "copyright year"

File: lib/ffmpeg/transcoderbase.py [hunk 1]
```diff
@@ -148,8 +156,12 @@ def gen_output_opts(self):
       filters = []
       tmode = (self.mode, mode)
       format = vars(self).get("format", "NV12")
-    
-      if ("hw", "sw") == tmode:   # HW to SW transcode
+
+      if self.tonemap in ["h2s"]:
+        tonemap = self.get_tonemap(self.tonemap, self.map_format(output.get("format", format)))
+        if tonemap is not None:
+          filters.append(tonemap)
```
[the remark this answers is not in the document]
Comment (xhaihao) [reply lib/ffmpeg/transcoderbase.py:163]: A separate filter is required for tonemap. Moreover YUV data will be changed a lot after tonemapping, so I don't want to use the original logic to check whether the result is expected.

**Comment by xhaihao** [comment]
> > @uartie @Bin-CI @FocusLuo Any comment for this PR ?
> 
> @xhaihao I gave a comment for "copyright year"

Thanks for review, fixed.

**Review by FocusLuo — APPROVED** [review]
LGTM, 
BTW, your PR only enable VME path for encode. I am not sure whether your tonemap filter supported LP mode? If supported, do you have any plan to add LP mode patch? (Because Gen12.7+ platforms only support LP encode mode)

**Comment by xhaihao** [comment]
> LGTM, BTW, your PR only enable VME path for encode. I am not sure whether your tonemap filter supported LP mode? If supported, do you have any plan to add LP mode patch? (Because Gen12.7+ platforms only support LP encode mode)

I didn't specify lp mode, ffmpeg will use LP mode automatically if vme isn't available. The main purpose of the tests is to verify whether HDR info is right after HDR to SDR tonemapping.

**Comment by FocusLuo** [comment]
> > LGTM, BTW, your PR only enable VME path for encode. I am not sure whether your tonemap filter supported LP mode? If supported, do you have any plan to add LP mode patch? (Because Gen12.7+ platforms only support LP encode mode)
> 
> I didn't specify lp mode, ffmpeg will use LP mode automatically if vme isn't available. The main purpose of the tests is to verify whether HDR info is right after HDR to SDR tonemapping.

vaapi-fits needs define VME or LP (this parameter is set by test cases full or smoke, in your PR in vaapi-fits-full, I only see VME - "HW" define), and in vaapi-fits this is enabled shown as below for ffmpeg. If not specify lp mode, this hevc h2s test cases will be skipped on Gen12.7+ platforms.
lib/ffmpeg/vaapi/transcoder.py
lib/ffmpeg/qsv/transcoder.py

**Comment by FocusLuo** [comment]
> > > LGTM, BTW, your PR only enable VME path for encode. I am not sure whether your tonemap filter supported LP mode? If supported, do you have any plan to add LP mode patch? (Because Gen12.7+ platforms only support LP encode mode)
> > 
> > 
> > I didn't specify lp mode, ffmpeg will use LP mode automatically if vme isn't available. The main purpose of the tests is to verify whether HDR info is right after HDR to SDR tonemapping.
> 
> vaapi-fits needs define VME or LP (this parameter is set by test cases full or smoke, in your PR in vaapi-fits-full, I only see VME - "HW" define), and in vaapi-fits this is enabled shown as below for ffmpeg. If not specify lp mode, this hevc h2s test cases will be skipped on Gen12.7+ platforms. lib/ffmpeg/vaapi/transcoder.py lib/ffmpeg/qsv/transcoder.py

You can have try your CMD on your platform (any platform is ok) only replace option with "--pl DG2", then you will see your H2S will be skipped

**Comment by xhaihao** [comment]
> > vaapi-fits needs define VME or LP (this parameter is set by test cases full or smoke, in your PR in vaapi-fits-full, I only see VME - "HW" define), and in vaapi-fits this is enabled shown as below for ffmpeg. If not specify lp mode, this hevc h2s test cases will be skipped on Gen12.7+ platforms. lib/ffmpeg/vaapi/transcoder.py lib/ffmpeg/qsv/transcoder.py
> 
> You can have try your CMD on your platform (any platform is ok) only replace option with "--pl DG2", then you will see your H2S will be skipped

Other hevc transcode cases are skipped too on DG2, e.g.  H_H-HEVC-ra_samsung_HDR
$ ./full run test --device /dev/dri/renderD129 --platform DG2 -k ffmpeg-vaapi -k H_H-HEVC-ra_samsung_HDR

I think we should update CAPS to allow user uses encoding without a given vme/lp mode.

**Comment by xhaihao** [comment]
> > You can have try your CMD on your platform (any platform is ok) only replace option with "--pl DG2", then you will see your H2S will be skipped
> 
> Other hevc transcode cases are skipped too on DG2, e.g. H_H-HEVC-ra_samsung_HDR $ ./full run test --device /dev/dri/renderD129 --platform DG2 -k ffmpeg-vaapi -k H_H-HEVC-ra_samsung_HDR

These cases are not skipped if applying https://github.com/xhaihao/vaapi-fits/tree/update-caps
> 
> I think we should update CAPS to allow user uses encoding without a given vme/lp mode.

**Pull request merged**